src/util/metadata/side_metadata/global.rs

```diff
@@ -146,7 +146,53 @@ impl SideMetadataSpec {
         MMAPPER.is_mapped_address(meta_addr)
     }
 
-    /// Bulk-zero a specific metadata for a chunk.
+    /// This method is used for bulk zeroing side metadata for a data address range. As we cannot guarantee
+    /// that the data address range can be mapped to whole metadata bytes, we have to deal with cases that
+    /// we need to mask and zero certain bits in a metadata byte.
+    /// The end address and the end bit are exclusive.
+    pub(super) fn zero_meta_bits(
+        meta_start_addr: Address,
+        meta_start_bit: u8,
+        meta_end_addr: Address,
+        meta_end_bit: u8,
+    ) {
+        // Start/end is the same, we don't need to do anything.
+        if meta_start_addr == meta_end_addr && meta_start_bit == meta_end_bit {
+            return;
+        }
+
+        // zeroing bytes
+        if meta_start_bit == 0 && meta_end_bit == 0 {
+            memory::zero(meta_start_addr, meta_end_addr - meta_start_addr);
+            return;
+        }
+
+        if meta_start_addr == meta_end_addr {
+            // we are zeroing selected bits in one byte
+            let mask: u8 = ((1 << meta_end_bit) - 1) - ((1 << meta_start_bit) - 1);
+            let mask = !mask;
```

Comment:
Can be simplified, although not as intuitive as the original one:
```suggestion
            let mask: u8 = (u8::MAX << meta_end_bit) | !(u8::MAX << meta_start_bit);
```

Reply:
Done